crates/noirc_evaluator/src/ssa_refactor/ir/dfg.rs

```diff
@@ -91,41 +100,54 @@ impl DataFlowGraph {
     }
 
     /// Inserts a new instruction into the DFG.
+    /// This does not add the instruction to the block or populate the instruction's result list
     pub(crate) fn make_instruction(&mut self, instruction_data: Instruction) -> InstructionId {
         let id = self.instructions.insert(instruction_data);
 
         // Create a new vector to store the potential results for the instruction.
         self.results.insert(id, Default::default());
+
         id
     }
 
+    pub(crate) fn make_allocate(&mut self, size: u32) -> (InstructionId, ValueId) {
+        let id = self.make_instruction(Instruction::Allocate { size });
+        self.make_instruction_results(id, Type::Reference);
+        (id, self.instruction_results(id)[0])
+    }
+
     pub(crate) fn make_value(&mut self, value: Value) -> ValueId {
         self.values.insert(value)
     }
 
-    /// Attaches results to the instruction.
+    pub(crate) fn constant(&mut self, value: FieldElement, typ: Type) -> ValueId {
```

Comment:
should this be called `make_constant` or is the make prefix reserved for a particular meaning?

Reply:
I didn't originally call it `make_constant` since it may just retrieve a constant after a lookup without "making" a new one, but perhaps `make_constant` would just be more consistent?

I was also trying to decide if our naming scheme should be `make_x`, `insert_x`, `push_x`, etc, or if each are for slightly different semantics. So far most things with instructions use insert

Reply:
Hmm thats a good point, the ConstantPool in Cranelift uses `insert` even though it may not insert, so `make_constant` or `make_or_get` would be the least suprising I think. 

For the naming scheme, I think we can bikeshed that later and we can go with whatever you think is best but consistent for now